animated_sprite.h

```diff
@@ -0,0 +1,94 @@
+#ifndef ANIMATEDSPRITE_H
+#define ANIMATEDSPRITE_H
+
+#include <SDL2/SDL.h>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "sprite.h"
+#include "vector2.h"
+
+class AnimatedSprite : public Sprite {
+ public:
+  // Constructor takes the name of the sprite sheet, the source rectangle for
+  // the first frame, the renderer, and the time between each frame.
+  AnimatedSprite(std::string sprite_sheet_name,
+                 SDL_Rect first_sprite_coord_location, SDL_Renderer* renderer,
+                 float time_to_update_sprite, SDL_Rect destination);
+
+  // Destructor.
+  virtual ~AnimatedSprite() = default;
+  // Checks to see if it is time to move to the next frame in the current
+  // animation.
+  void TimeToUpdateFrame();
```

Comment:
This method strikes me as odd. It returns nothing and it isn't clear what it does.

Try a better comment and name.

Reply:
fix